server/src/main/java/org/elasticsearch/rest/action/admin/cluster/RestNodesHotThreadsAction.java

```diff
@@ -34,22 +36,42 @@
 
 import java.io.IOException;
 import java.util.List;
+import java.util.stream.Collectors;
 
 import static org.elasticsearch.rest.RestRequest.Method.GET;
 
-
 public class RestNodesHotThreadsAction extends BaseRestHandler {
 
+    private static final DeprecationLogger deprecationLogger = new DeprecationLogger(
+        LogManager.getLogger(RestNodesHotThreadsAction.class));
+    static final String DEPRECATED_MESSAGE_CLUSTER_NODES_HOTTHREADS = "[/_cluster/nodes/hotthreads] is a deprecated endpoint. " 
+        + "Please use [/_cluster/nodes/hot_threads] instead.";
+    static final String DEPRECATED_MESSAGE_CLUSTER_NODES_NODEID_HOTTHREADS = "[/_cluster/nodes/{nodeId}/hotthreads] is a deprecated endpoint. "
+        + "Please use [/_cluster/nodes/{nodeId}/hot_threads] instead.";
+    static final String DEPRECATED_MESSAGE_NODES_HOTTHREADS = "[/_nodes/hotthreads] is a deprecated endpoint. "
+        + "Please use [/_nodes/hot_threads] instead.";
+    static final String DEPRECATED_MESSAGE_NODES_NODEID_HOTTHREADS = "[/_nodes/{nodeId}/hotthreads] is a deprecated endpoint. "
+        + "Please use [/_nodes/{nodeId}/hot_threads] instead.";
+
+    @Override
+    public List<DeprecatedRoute> deprecatedRoutes() {
+        return List.of(
+            new DeprecatedRoute(GET, "/_cluster/nodes/hotthreads",
+                    DEPRECATED_MESSAGE_CLUSTER_NODES_HOTTHREADS),
+            new DeprecatedRoute(GET, "/_cluster/nodes/{nodeId}/hotthreads",
+                    DEPRECATED_MESSAGE_CLUSTER_NODES_NODEID_HOTTHREADS),
+            new DeprecatedRoute(GET, "/_nodes/hotthreads",
+                    DEPRECATED_MESSAGE_NODES_HOTTHREADS),
+            new DeprecatedRoute(GET, "/_nodes/{nodeId}/hotthreads",
+                    DEPRECATED_MESSAGE_NODES_NODEID_HOTTHREADS));
```

Comment:
Please avoid using non-const string, maybe consider to make repeatedly string const.

Reply:
I see! Thanks :)